File: src/main/java/io/github/chindeaone/collectiontracker/collections/CollectionsManager.java
package io.github.chindeaone.collectiontracker.collections;

import java.util.*;

public class CollectionsManager {

    public static Map<String, Set<String>> collections = new HashMap<>();
    public static String collectionSource;
    public static String collectionType;

    public static List<String> multiCollectionSource = new LinkedList<>();
    public static Map<String, String> multiCollectionTypes = new HashMap<>();

    @SuppressWarnings("BooleanMethodIsAlwaysInverted")
    public static boolean isValidCollection(String collectionName) {
        for (Set<String> collectionSet : collections.values()) {
            if (collectionSet.contains(collectionName)) {
                return true;
            }
        }
        return false;
    }

    public static boolean isCollection(String collectionName) {
        for (Map.Entry<String, Set<String>> entry : collections.entrySet()) {
            if (!entry.getKey().equals("Miscellaneous") && entry.getValue().contains(collectionName)) {
                return true;
            }
        }
        return false;
    }

    public static List<String> getAllCollections() {
        List<String> allCollections = new LinkedList<>();
        for (Set<String> collectionSet : collections.values()) {
            allCollections.addAll(collectionSet);
        }
        return allCollections;
    }

    public static boolean isRiftCollection(String collectionName) {
        return collections
                .getOrDefault("Rift", Collections.emptySet())
                .contains(collectionName);
    }

    public static void resetCollections() {
        collectionSource = null;
        collectionType = null;
    }

    public static void resetMultiCollections() {
        multiCollectionSource.clear();
        multiCollectionTypes.clear();
    }
}

File: src/main/java/io/github/chindeaone/collectiontracker/collections/CollectionsManager.kt
package io.github.chindeaone.collectiontracker.collections

import io.github.chindeaone.collectiontracker.commands.CollectionTracker.collectionList

object CollectionsManager {
    var collections = linkedMapOf<String, Set<String>>()
    var collectionSource: String? = null
    var collectionType: String? = null

    var multiCollectionSource: MutableList<String> = mutableListOf()
    var multiCollectionTypes: MutableMap<String, String> = mutableMapOf()

    fun isValidCollection(collectionName: String): Boolean {
        for (collectionSet in collections.values) {
            if (collectionSet.contains(collectionName)) {
                return true
            }
        }
        return false
    }

    fun isCollection(collectionName: String): Boolean {
        for (entry in collections.entries) {
            if (entry.key != "Miscellaneous" && entry.value.contains(collectionName)) {
                return true
            }
        }
        return false
    }

    val allCollections: MutableList<String>
        get() {
            val allCollections: MutableList<String> = mutableListOf()
            for (collectionSet in collections.values) {
                allCollections.addAll(collectionSet)
            }
            return allCollections
        }

    fun isRiftCollection(collectionName: String): Boolean {
        return collections
            .getOrDefault("Rift", mutableSetOf())
            .contains(collectionName)
    }

    fun hasAnyRiftCollection(): Boolean {
        val riftCollections = collections.getOrDefault("Rift", mutableSetOf())

        return collectionList.stream().anyMatch { o: String? -> riftCollections.contains(o) }
    }

    fun hasAllRiftCollections(): Boolean {
        val riftCollections = collections.getOrDefault("Rift", mutableSetOf())

        return riftCollections.containsAll(collectionList)
    }

    fun resetCollections() {
        collectionSource = null
        collectionType = null
    }

    fun resetMultiCollections() {
        multiCollectionSource.clear()
        multiCollectionTypes.clear()
    }
}